presidio_evaluator/evaluation/span_evaluator.py

```diff
@@ -477,12 +402,16 @@ def calculate_score(
         If None, all entities are considered.
         :param beta: The beta parameter for F-beta score calculation. Default is 2.
         """
-
+        evaluation_result = EvaluationResult()
         df = self.get_results_dataframe(evaluation_results=evaluation_results, entities=entities)
-        return self.calculate_score_on_df(df, beta=beta)
+        evaluation_result = self.calculate_score_on_df(per_type=True, results_df=df, beta=beta, evaluation_result=evaluation_result)
+        global_pii_df = self.create_global_entities_df(results_df=df)
+        evaluation_result = self.calculate_score_on_df(per_type=False, results_df=global_pii_df, beta=beta, evaluation_result=evaluation_result)
+        return evaluation_result
 
     def calculate_score_on_df(
-        self, results_df: pd.DataFrame, beta: float = 2
+        self, per_type: bool, results_df: pd.DataFrame, beta: float = 2, 
+        evaluation_result: EvaluationResult = EvaluationResult, 
```

Comment:
```suggestion
        evaluation_result: Optional[EvaluationResult] = None,
```
if the user doesn't pass an EvaluationResult, we should create one. Otherwise it might not be clear what this parameter should contain.